azerothcore/azerothcore-wotlk
# fix(Scripts/Mail) Fix faction server mail rewards.

## Changes Proposed:
-  Switches Horde and Alliance rewards in ObjectMgr.cpp.

## Issues Addressed:

- Closes #12592 

## SOURCE:


## Tests Performed:

- Tested.
```
INSERT INTO `mail_server_template` (`id`, `reqLevel`, `reqPlayTime`, `moneyA`, `moneyH`, `itemA`, `itemCountA`, `itemH`, `itemCountH`, `subject`, `body`, `active`) VALUES
(1, 1, 1, 20, 10, 0, 0, 0, 0, 'Big gamer (Alliance)', '20 gold for alliance, 10 for horde', 1),
(2, 1, 1, 10, 20, 0, 0, 0, 0, 'Big gamer (Horde)', '20g Horde, 10g ally', 1);
```
![image](https://user-images.githubusercontent.com/38561765/182009954-83d62533-b576-423c-a986-ebc261d65b50.png)
![image](https://user-images.githubusercontent.com/38561765/182009956-e1fb2136-610d-4a49-a7d3-4edfb8710f9a.png)



## How to Test the Changes:


1. Run the query seen above.
2. Create a character, log in, and restart the worldserver.
3. Observe the correct money being given to your character upon logging back in.

## Known Issues and TODO List:


- Nope.
Credit for the fix goes entirely to @AnchyDev <3


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
